components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -3574,6 +3595,35 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_year_as_string() {
+        test_none_with_ctx_and_extra(cast_year_as_string);
+
+        let cs: Vec<(i64, Vec<u8>, String)> = vec![
+            (
+                0,
+                "0000".as_bytes().to_vec(),
+                "0000".to_string(),
+            ),
+            (
+                2000,
+                "2000".as_bytes().to_vec(),
+                "2000".to_string(),
+            ),
+        ];
+
+        let ref_cs = helper_get_cs_ref(&cs);
+
+        test_as_string_helper(
+            ref_cs,
+            |ctx, extra, val| {
+                let val = val.map(|x| *x as i64);
+                cast_uint_as_string(ctx, extra, val.as_ref())
+            },
+            "cast_year_as_string",
```

Comment:
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/36921/pipeline#step-97-log-39

`cargo fmt` is required here.

Reply:
done